### PR TITLE
Feat/83/execute decree operation

### PR DESCRIPTION
**Pull request by Zakerman110**
Ability to transfer soldiers between divisions according to decree

### CODE REVIEW

File: MilitaryManager.Core/Entities/TemplatePlaceholderEntity/TemplatePlaceholder.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int TemplateId { get; set; }
+        public Template Template { get; set; }
+        public ICollection<DecreeData> DecreeDatas { get; set; }
+
+
+        public List<TemplatePlaceholder> GetTemplatePlaceholders()
+        {
+            List<TemplatePlaceholder> placeholders = new List<TemplatePlaceholder>()
```
Comment (Alex-knu) [inline MilitaryManager.Core/Entities/TemplatePlaceholderEntity/TemplatePlaceholder.cs]: Move data to Seed file

File: MilitaryManager.Core/Entities/TemplatePlaceholderEntity/TemplatePlaceholder.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public class TemplatePlaceholder : IBaseEntity<int>
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int TemplateId { get; set; }
+        public Template Template { get; set; }
+        public ICollection<DecreeData> DecreeDatas { get; set; }
+
+
+        public List<TemplatePlaceholder> GetTemplatePlaceholders()
```
Comment (Alex-knu) [inline MilitaryManager.Core/Entities/TemplatePlaceholderEntity/TemplatePlaceholder.cs]: use db

**Review by Alex-knu — CHANGES_REQUESTED** [review]

**Review by Zhykovskiy — APPROVED** [review]

File: MilitaryManager.Core/Entities/UnitEntity/Units.cs [hunk 3]
```diff
@@ -54,5 +54,12 @@ public UnitById(int id)
                 .Where(x => x.Id == id);
             }
         }
+        internal class UnitByFirstLastSecondNames : Specification<Unit>
+        {
+            public UnitByFirstLastSecondNames(string firstName, string lastName, string secondName)
+            {
+                Query.Where(x => x.FirstName == firstName || x.LastName == lastName || x.SecondName == secondName);
```
Comment (Alex-knu) [inline MilitaryManager.Core/Entities/UnitEntity/Units.cs]: use &&

**Review by Alex-knu — CHANGES_REQUESTED** [review]

File: MilitaryManager.Core/Entities/DecreeDataEntity/DecreeDatas.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            {
+                Query.Where(x => x.TemplatePlaceholderId == placeholderId && x.DecreeId == decreeId);
+            }
+        }
+
+        internal class DecreeDataUnitCredDivisionNumber : Specification<DecreeData>
+        {
+            public DecreeDataUnitCredDivisionNumber(int firstNameId, int lastNameId, int secondNameId, int divisionNumberId, int decreeId)
+            {
+                Query.Where(x => x.TemplatePlaceholderId == firstNameId && x.DecreeId == decreeId ||
```
Comment (Alex-knu) [inline MilitaryManager.Core/Entities/DecreeDataEntity/DecreeDatas.cs]: (x.TemplatePlaceholderId == firstNameId || x.TemplatePlaceholderId == lastNameId || x.TemplatePlaceholderId == secondNameId || x.TemplatePlaceholderId == divisionNumberId)  && x.DecreeId == decreeId

**Review by Alex-knu — CHANGES_REQUESTED** [review]

**Review by Alex-knu — APPROVED** [review]

**Pull request merged**